iravelmakina/eigenvectors-ivelmakina
# Part two

### Description:
- Added functionality to load, display, and process an image using PCA.
- Converted the image to black and white.
- Applied PCA to the image data.
- Displayed cumulative variance and determined the number of components needed to cover 95% of the variance.
- Reconstructed the image using the determined number of components.
- Reconstructed the image for various numbers of components and displayed the results.

### Changes:
- Added code to load and display the original image.
- Converted the image to black and white and displayed it.
- Applied PCA and calculated cumulative variance.
- Determined the number of PCA components needed to cover 95% of the variance.
- Reconstructed the image using the calculated number of components.
- Added functionality to reconstruct the image using different numbers of components and display the results.

### How to Test:
1. Load and run the provided script.
2. Verify that the original image is displayed correctly.
3. Check the black and white conversion of the image.
5. Ensure that the PCA cumulative variance is found and plotted correctly.
6. Confirm that the number of components covering 95% of the variance is correct.
7. Verify that the reconstructed image using the limited number of components is displayed correctly.
8. Check the reconstructed images for different numbers of components to ensure they are displayed correctly.

### Screenshots:
*Original Image*
![image](https://github.com/iravelmakina/eigenvectors-ivelmakina/assets/150923968/ee703b70-ec4a-43c5-85cb-b23c48c8112d)

*Black and White Image*
![image](https://github.com/iravelmakina/eigenvectors-ivelmakina/assets/150923968/ad376f77-bfa3-49c3-9672-8a730a32d4c4)

*Cumulative variance and Number of components to cover 95% of the variance*
![image](https://github.com/iravelmakina/eigenvectors-ivelmakina/assets/150923968/d4b0d258-edef-427c-ab80-f1d5ffc08cd0)
![image](https://github.com/iravelmakina/eigenvectors-ivelmakina/assets/150923968/6a25ed91-c987-4a8d-8892-5a6337f33181)
![image](https://github.com/iravelmakina/eigenvectors-ivelmakina/assets/150923968/e9780e2d-f6ac-441e-bfe8-3b3f212600b1)

*Cumulative Variance Plot*
![image](https://github.com/iravelmakina/eigenvectors-ivelmakina/assets/150923968/691edc3a-3673-48ea-b7ab-5ff22b45fef1)

*Reconstructed Image using limited components*
![image](https://github.com/iravelmakina/eigenvectors-ivelmakina/assets/150923968/ddd1f049-41d5-44c1-bb13-59105202ebee)

*Reconstructed Images using different numbers of components*
![image](https://github.com/iravelmakina/eigenvectors-ivelmakina/assets/150923968/c57573a4-cc3e-4474-8254-b20c5d9b4a9e)
